grease/src/Grease/Heuristic.hs

```diff
@@ -121,6 +122,12 @@ refinePtrArg ::
   IO (HeuristicResult ext argTys)
 refinePtrArg la args modify sel =
   case args ^. selectArg sel of
+    ShapeExt (ShapePtrBV _tag w) | Just Refl <- testEquality w ?ptrWidth -> do
+      let pt = ptrTarget Nothing Seq.empty
+      doLog la $ Diag.HeuristicPtrTarget pt
+      pt' <- modify pt
+      let args' = args & selectArg sel .~ ShapeExt (ShapePtr NoTag (Offset 0) pt')
```

Comment:
Im trying to work out if this matches up with the "heuristics wont make empty memshapes" piece. Im assuming that the modify provided is always supposed to grow this target allocation, is that always true?

Reply:
Yeah, in practice this is always one of the `.*(g|G)rowPtrTarget.*` functions.